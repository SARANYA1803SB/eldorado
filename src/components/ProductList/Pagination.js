import React, { useState } from 'react';
import { Container, Row, Col, Image } from 'react-bootstrap';
import './ProductList.css';

const Pagination = (props) => {


    const [pages] = useState(Math.ceil(props.totalPosts / props.postsPerPage));
    const [currentPage, setCurrentPage] = useState(1);

    //Get current posts
    const indexOfLastPost = currentPage * props.postsPerPage;
    const indexOfFirstPost = indexOfLastPost - props.postsPerPage;
    const currentPosts = props.productsDetail.slice(indexOfFirstPost, indexOfLastPost);

    function goToNextPage() {
        setCurrentPage((page) => page + 1);
    }

    function goToPreviousPage() {
        setCurrentPage((page) => page - 1);
    }
    const getPaginationGroup = () => {
        if (pages < props.pageLimit) {

            let start = Math.floor((currentPage - 1) / pages) * pages;
            return new Array(pages).fill().map((_, idx) => start + idx + 1);

        } else if (pages >= props.pageLimit && currentPage <= props.pageLimit) {

            let start = Math.floor((currentPage - 1) / props.pageLimit) * props.pageLimit;
            return new Array(props.pageLimit).fill().map((_, idx) => start + idx + 1);

        } else if (pages > props.pageLimit && currentPage > props.pageLimit) {

            let start = currentPage - 1;
            return new Array(props.pageLimit).fill().map((_, idx) => start + idx - 3);

        }
    };

    function changePage(event) {
        const pageNumber = Number(event.target.textContent);
        setCurrentPage(pageNumber);
    }


    const product = (productData) => {
        return (
            <div className="col-6 col-md-4">
                <div className="card shadow p-3 mb-5 bg-white rounded">
                    <Col className="center" xs={6} md={4}>
                        <Image src={productData.imageUrl} thumbnail />
                    </Col>
                    <p className="title">Name :{productData.name}</p>
                    <p className="price">Price :{productData.price}</p>
                </div>
            </div>
        );
    };

    return (
        <div>
            <Container>
                <Row>
                    {currentPosts.map(product)}
                </Row>

                <nav className="nav justify-content-center">
                    <ul className="pagination">
                        <button
                            onClick={goToPreviousPage}
                            className={`page-link ${currentPage === 1 ? 'disabled' : ''}`}
                        >
                            prev
                        </button>
                        {getPaginationGroup().map((item, index) => (
                            <li key={index} className="page-item">
                                <button onClick={changePage} className={`page-link ${currentPage === item ? 'active' : null}`}>
                                    {item}
                                </button>
                            </li>
                        ))}
                        <button
                            onClick={goToNextPage}
                            className={`page-link ${currentPage === pages ? 'disabled' : ''}`}
                        >
                            next
                        </button>

                    </ul>
                </nav>
            </Container>

        </div>
    );
}

export default Pagination;