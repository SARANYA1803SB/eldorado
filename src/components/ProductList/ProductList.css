.thumbnail {
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.225);
  width: 350px;
  height: 266px;
  margin: 1%;
  /* border:1px solid black; */
}
.thumbnail:hover {
  box-shadow: 0 8px 16px 0 rgba(0, 0, 0, 0.225);
}
.card {
  box-shadow: 0 8px 16px 0 rgba(0, 0, 0, 0.225);
}
.card:hover {
  transform: scale(1.1);
  opacity: 1.5;
  transition: 0.3s;
}
.center {
  margin: auto;
}
.price {
  text-align: left;
  color: red;
  font-size: 1.25rem;
}
.title {
  text-align: left;
  color:steelblue;
  font-size: 1.25rem;
}

.disabled {
  pointer-events: none;
  box-shadow: none;
  color: #999;
}

.active {
  border: 2px solid black;
  color: #888;
  pointer-events: none;
}
