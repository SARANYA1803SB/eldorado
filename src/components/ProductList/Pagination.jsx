import React, { useState } from 'react';
import './ProductList.css';

const Pagination = (props) => {


    const [pages] = useState(Math.ceil(props.totalPosts / props.postsPerPage));
    
    function goToNextPage() {
        props.paginate(props.currentPage+1);
    }

    function goToPreviousPage() {
        props.paginate(props.currentPage-1);
    }
    
    const getPaginationGroup = () => {
        if (pages < 5) {

            let start = Math.floor((props.currentPage - 1) / pages) * pages;
            return new Array(pages).fill().map((_, idx) => start + idx + 1);

        } else if (pages >= 5 && props.currentPage <= 5) {

            let start = Math.floor((props.currentPage - 1) / 5) * 5;
            return new Array(5).fill().map((_, idx) => start + idx + 1);

        } else if (pages > 5 && props.currentPage > 5) {

            let start = props.currentPage - 1;
            return new Array(5).fill().map((_, idx) => start + idx - 3);

        }
    };

    function changePage(event) {
        const pageNumber = Number(event.target.textContent);
        props.paginate(pageNumber);
    }


    

    return (
    
                <nav className="nav justify-content-center">
                    <ul className="pagination">
                        <button
                            onClick={goToPreviousPage}
                            className={`page-link ${props.currentPage === 1 ? 'disabled' : ''}`}
                        >
                            prev
                        </button>
                        {getPaginationGroup().map((item, index) => (
                            <li key={index} className="page-item" placeholder="number value">
                                <button onClick={changePage} className={`page-link ${props.currentPage === item ? 'active' : null}`} placeholder="buttonElement">
                                    {item}
                                </button>
                            </li>
                        ))}
                        <button
                            onClick={goToNextPage}
                            className={`page-link ${props.currentPage === pages ? 'disabled' : ''}`}
                        >
                            next
                        </button>

                    </ul>
                </nav>
           
    );
}

export default Pagination;