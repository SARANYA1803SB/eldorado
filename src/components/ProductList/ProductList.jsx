import React, { useState } from 'react';
import { Container, Row, Col, Image } from 'react-bootstrap';
import Pagination from './Pagination';
import { productsDetail } from './../../utils/ProductListUtil';
import './ProductList.css';

const ProductList = (props) => {

    const[currentPage,setCurrentPage]=useState(1);
    const [postsPerPage]=useState(9);

    //Get current posts
    const indexOfLastPost=currentPage*postsPerPage;
    const indexOfFirstPost=indexOfLastPost-postsPerPage;
    const currentPosts=productsDetail.slice(indexOfFirstPost,indexOfLastPost); 

    // Change page
    const paginate =(number) =>setCurrentPage(number);
    
    const product = (productData) => {
        return (
            <div className="col-6 col-md-4" > 
            <div className="card p-3 mb-5 bg-white rounded thumbnail" id="item">
                <Col className="center" xs={6} md={4}>
                <Image  src={productData.imageUrl} card="true"/>
                </Col>
                <p className="title" id="title"><a>Name :{productData.name}</a></p>
                <p className="price" id="price">Price :{productData.price}</p>
            </div>
            </div>

        );
    };

    return (
        <div>
            <Container>
                <Row>
                {currentPosts.map(product)}
                </Row>
                <Pagination
                postsPerPage={postsPerPage}
                totalPosts={productsDetail.length}
                paginate={paginate}
                currentPage={currentPage}
                />
            </Container>
            
        </div>
        
        
    );
}

export default ProductList;