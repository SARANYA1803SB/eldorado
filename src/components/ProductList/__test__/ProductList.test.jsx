import {configure,shallow} from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import React from 'react';
import { render } from "@testing-library/react";
import ProductList from '../ProductList';
import {paginate} from "../ProductList";


configure({ adapter: new Adapter() });

  describe("Product List Page", () => {
    let wrapper;
    beforeEach(() => {
        // setup a DOM element as a render target
        wrapper = shallow(<ProductList />);
      });

    it("Test Product Title in Product List", () => {
        expect(wrapper).toMatchSnapshot();
    });
    it("Test Product Title in Product List", () => {
        expect(wrapper.find("#title").first().text()).toBe("Name :Minions");
      });
    it("Test Product Title in Product List", () => {
        expect(wrapper.find("#price").first().text()).toBe("Price :1000");
      });
      
    it("Test Product Title in Product List", () => {
        let imageUrl= "https://i.pinimg.com/originals/c7/e6/42/c7e64283cd631ebd281e23fc47d0316e.jpg";
        expect(wrapper.find("Image").first().prop("src")).toEqual(imageUrl);
    });

    it("Test Product Title in Product List", () => {
        let itemsPerPage=9;
         expect(wrapper.find('#item').children()).toHaveLength(itemsPerPage*3);
    });
    //it('Set Page Number', () => {
    //  expect(paginate(1)).toMatchSnapshot();
    //});

    
    
  });