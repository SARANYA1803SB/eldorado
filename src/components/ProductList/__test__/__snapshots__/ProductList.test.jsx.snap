// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product List Page Test Product Title in Product List 1`] = `ShallowWrapper {}`;
